crates/ide-assists/src/handlers/inline_call.rs

```diff
@@ -218,13 +219,12 @@ pub(crate) fn inline_call(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<
     }
 
     let syntax = call_info.node.syntax().clone();
+    let replacement = inline(&ctx.sema, file_id, function, &fn_body, &params, &call_info)?;
```

Comment:
> So while I do not claim to understand why the assist became non-lazy after this PR

Assists run in two modes: once to determine if they are available, and once to actually apply the changes, if the user picks one. Both modes use mostly the same code paths, but the `|builder|` closure actually determines the changes. This means that everything that happens outside that closure is performance-sensitive.

Reply:
thanks for the explanation